mxcube3/ui/actions/sampleChanger.js

```diff
@@ -65,20 +98,20 @@ export function scan(address) {
       credentials: 'include'
     }).then(response => {
       if (response.status >= 400) {
-        dispatch(setState('ERROR'));
+        // dispatch(setSCState('ERROR'));
         throw new Error(`Error while scanning sample changer @ ${address}`);
       }
 
       response.json().then(contents => { dispatch(setContents(contents)); });
 
-      dispatch(setState('READY'));
+      // dispatch(setSCState('READY'));
     });
   };
 }
 
 export function loadSample(address) {
   return function (dispatch) {
-    dispatch(setState('MOVING'));
```

Comment:
Why is 'MOVING' state dispatching removed ? We think it is better to immediately reflect that we are doing something with the sample changer, instead of waiting for the next state update. Indeed, it would make it more responsive for users and it has the added benefit of preventing crazy 'clickers' to send many commands while one has already been sent.
We can ask ourselves the same question for the READY state at the end, although it is not so clear because theoretically the last updated state is the right one, and in this case the new code seems more correct.